compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/MetalRedrawer.uikit.kt

```diff
@@ -218,6 +218,9 @@ internal class MetalRedrawer(
         target = DisplayLinkProxy {
             val targetTimestamp = currentTargetTimestamp ?: return@DisplayLinkProxy
 
+            // Run all scheduled operations in current Run Loop first, then do rendering.
+            NSRunLoop.currentRunLoop().runUntilDate(NSDate())
```

Comment:
I have hard times imagining how some work we are actually interested in dispatching now can appear here. Both DisplayLink and Compose dispatcher are bound to main thread currently. Can you explain a sequence of events which is fixed by this change?

Reply:
Explanation would be helpful.

Also, if it is related to the Compose core, the fix should be in `ComposeScene`, as the other targets may have the same issue. We probably need to reuse `FlushCoroutineDispatcher` as I mentioned in the Slack thread.

If it is not related to the core, then it most probably related only to `MetalRedrawer`, and in this case, it is better to not flush the whole main dispatcher, but only to flush the task scheduled for the redrawer.

Reply:
Removed.